src/unimport/main.py

```diff
@@ -1,90 +1,146 @@
-import sys
-from typing import Optional, Sequence
+import contextlib
+import dataclasses
+from pathlib import Path
+from typing import TYPE_CHECKING, Iterator, List, Optional, Sequence, Union
 
-from unimport import color, commands, emoji, utils
+from unimport import color, commands, utils
 from unimport.analyzer import Analyzer
-from unimport.commands import generate_parser
-from unimport.config import ParseConfig
-from unimport.refactor import refactor_string
+from unimport.config import Config
 from unimport.statement import Import
-from unimport.utils import return_exit_code
 
-__all__ = ("main",)
+if TYPE_CHECKING:
+    from unimport.statement import ImportFrom
 
 
-def main(argv: Optional[Sequence[str]] = None) -> int:
-    config = ParseConfig.parse_args(
-        generate_parser().parse_args(
-            argv if argv is not None else sys.argv[1:]
-        )
-    )
-    unused_import_names, is_syntax_error, refactor_applied = (
-        set(),
-        False,
-        False,
+__all__ = ("Main",)
+
+
+@dataclasses.dataclass
+class _Result:
+    unused_imports: List[Union["Import", "ImportFrom"]] = dataclasses.field(
+        repr=False
     )
-    for path in config.get_paths():
-        source, encoding, newline = utils.read(path)
+    path: Path
+    source: str
+    encoding: str
+    newline: Optional[str] = None
+
+
+@dataclasses.dataclass
+class Main:
+    argv: Optional[Sequence[str]] = None
+
+    config: Config = dataclasses.field(init=False)
+    is_syntax_error: bool = dataclasses.field(init=False, default=False)
+    is_unused_imports: bool = dataclasses.field(init=False, default=False)
+    refactor_applied: bool = dataclasses.field(init=False, default=False)
+
+    def __post_init__(self):
+        self.config = self.argv_to_config()
 
+    def argv_to_config(self) -> Config:
+        import sys
+
+        from unimport.config import ParseConfig
+
+        return ParseConfig.parse_args(
+            commands.generate_parser().parse_args(
+                self.argv if self.argv is not None else sys.argv[1:]
+            )
+        )
+
+    @contextlib.contextmanager
+    def analysis(self, source: str, path: Path) -> Iterator:
         analysis = Analyzer(
             source=source,
             path=path,
-            include_star_import=config.include_star_import,
+            include_star_import=self.config.include_star_import,
         )
         try:
             analysis.traverse()
         except SyntaxError as exc:
             print(
-                color.paint(str(exc), color.RED, config.use_color)
+                color.paint(str(exc), color.RED, self.config.use_color)
                 + " at "
-                + color.paint(path.as_posix(), color.GREEN, config.use_color)
+                + color.paint(
+                    path.as_posix(), color.GREEN, self.config.use_color
+                )
             )
-            is_syntax_error = True
-            continue
+            if self.is_syntax_error is False:
```

Comment:
There is no need to if statement. You can use ```self.is_syntax_error = True```

Reply:
fix it